book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -61,7 +62,70 @@ state service.
 # Guide-level explanation
 [guide-level-explanation]: #guide-level-explanation
 
-XXX fill in after writing other details
+The `zebra-state` crate provides an implementation of the chain state storage
+logic in a zcash consensus node. Its main responsibility is to store chain
+state, validating new blocks against the existing chain state in the process,
+and to allow later querying of said chain state. `zebra-state` provides this
+interface via a `tower::Service` based on the actor model with a
+request/response interface for passing messages back and forth between the
+state service and the rest of the application.
+
+The main entry point for the `zebra-state` crate is the `init` function. This
+function takes a `zebra_state::Config` and constructs a new state service,
+which it returns wrapped by a tower::Buffer. This service is then interacted
+with via the `tower::Service` trait.
+
+```rust
+use tower::{Service, ServiceExt};
+
+let config = app_config();
+let state_config = config.state;
+let network = config.network;
+
+let state = zebra_state::on_disk::init(state_config, network);
+let request = zebra_state::Request::GetBlockLocator { genesis: genesis_hash };
+let response = state.ready_and().await?.call(request).await?;
+
+assert!(matches!(response, zebra_state::Response::BlockLocator(_)));
+```
+
+**Note**: The `tower::Service` API requires that `ready` is always called
+exactly once before each `call`. It is up to users of the zebra state service
+to uphold this contract.
+
+The service itself is clonable. When cloned it only clones the buffered
+interface, and not the wrapped service, providing shared access to a common
+chain state across multithreaded applications.
```

Comment:
Rather than say "The service itself is clonable" (which isn't technically true, because the service itself *isn't* clonable), this should say that the `Buffer` wrapper is clonable, sharing access to the inner service:

```suggestion
The `tower::Buffer` wrapper is `Clone`able, allowing shared access to a common state service.  This allows different tasks to share access to the chain state.
```

Talking about tasks is better than threads IMO because the unit of work in any application using `zebra-state` is a task, not a thread.